packages/react-popup/Spec.md

```diff
@@ -1,31 +1,385 @@
+
+### Position/Alignment hints
+
+Both libraries provide an API that achieves the same end result for positioning and alignment. Below is a table that maps the v8 `DirectionalHint` with the v0 props of `position` and `alignment`
+
+| DirectionalHint (v7) | position (v0) | align (v0) |
+| -------------------- | ------------- | ---------- |
+| topLeftEdge          | above         | start      |
+| topCenter            | above         | center     |
+| topRightEdge         | above         | bottom     |
+| topAutoEdge          | above         |            |
+| bottomLeftEdge       | above         | start      |
+| bottomCenter         | below         | center     |
+| bottomRightEdge      | below         | bottom     |
+| bottomAutoEdge       | below         |            |
+| leftTopEdge          | before        | top        |
+| leftCenter           | before        | center     |
+| leftBottomEdge       | before        | bottom     |
+| rightTopEdge         | after         | before     |
+| rightCenter          | after         | center     |
+| rightBottomEdge      | after         | bottom     |
+
+v8 uses `left` and `right`. v0 uses `before` and `after`. v0 vocabulary tries to be consistent regardless of RTL state. It's also possible to supply an explicit RTL hint to v8 which is a flip by default. v0 will flip by default but requires the consumer to detect RTL scenarios and modify props in these situations.
+
+In general the separation of both the position and alignment in v0 results in an API that is easier to use if a consumer only needs to modify one of the two props. However both try to achieve the same result in the end.
+
+It's important to note that if an incorrect pair of `position` and `align` are provided in v0, then `position` takes priority and `align` is set to `center`
+
+### Offset
+
+```tsx
+<ContextualMenu
+  // single number value
+  gap={100}
+/>
+
+<Popup
+  offset={[-100, 100]}
+/>
+
+// offset can also be a function of raw Popper properties
+const offsetFunction = ({
+  popper: PopperJs.Rect;
+  reference: PopperJs.Rect;
+  placement: PopperJs.Placement;
+}) => ([popper.width, -popper.height])
+```
+
+v8 positioning can only apply a numerical value to the first part position attribute of `DirectionalHint`. v0 supports a function to defer calculation at runtime. v0 also supports offset of the `Popup` in both axes while supporting RTL flips for offset values.
+
+### Bounds and overflow
+
+```typescript
+<Popup
+...
+  flipBoundary={htmlElement}
+  overflowBoundary={htmlElement}
+  mountNode={htmlElement}
+/>
+```
+
+v0 `Popup` provides 3 different properties to handle bounds and overflow:
+
+- flipBoundary - the bounds to calculate when to flip positioning of the popup
+- overflowBoundary - the bounds to shift the popup without overflowing
+- mountNode - where the popup is actually rendered in the DOM, by default this is a portal to a div in body
+
+```typescript
+<ContextualMenu
```

Comment:
Should this be `Callout` instead of `ContextualMenu`?

Reply:
good point, I wrote up this stuff during the initial menu days